src/PFRICH_geo.cpp

```diff
@@ -113,11 +111,16 @@ static Ref_t createDetector(Detector& description, xml_h e, SensitiveDetector se
   double radiatorRmin = radiatorElem.attr<double>(_Unicode(rmin));
   double radiatorRmax = radiatorElem.attr<double>(_Unicode(rmax));
 
+  auto filterElem = radiatorElem.child(_Unicode(filter));
+
   double airgapThickness = 0.1;
   double filterThickness = 1;
 
-  auto aerogelMat = description.material("C4F10_PFRICH");
-  auto filterMat  = description.material("C4F10_PFRICH");
+  auto aerogelMat = description.material(aerogelElem.attr<std::string>(_Unicode(material)));
+  auto filterMat  = description.material(filterElem.attr<std::string>(_Unicode(material)));
+
+  //  auto aerogelMat = description.material("C4F10_PFRICH");
+  //  auto filterMat  = description.material("C4F10_PFRICH");
```

Comment:
```suggestion
```

The change is recorded in the git history. You can use `git log -p -- src/PFRICH_geo.cpp` to access relevant changes.